### ISSUE TITLE
Cash Account Integration

### ISSUE THREAD

**Issue by monsterbitar** [Bitcoin-com/Wallet#452]
I am currently working on a specification for on-chain aliases that look like this: James#577821; to help convey payment information in a more convenient manner, particulary in verbal contexts.

The specification has undergone reasonable peer-review so far (by wallet developers, token developers, coinbase engineeers) but more feedback is always welcome.

Here's a link to the draft spec: https://gitlab.com/cash-accounts/specification/blob/master/SPECIFICATION.md

**Comment by coiki** [comment]
Looks very interesting! We'll definitely look into it. Thanks, great initiative.

**Comment by Joshua678891011** [comment]
Great idea

**Comment by Bitkilo** [comment]
It would be great to be able to "tag" the address used for your Cash Account
I got mine :)

**Comment by hfsam** [comment]
我的货币和现金在哪裡？

**Comment by 3ggerman29** [comment]
kindly Deposite on btcash wallet address  bitcoincash:qprtuvakg7na6qdjf5alypgjrvcm6qj6nv0dwnpvsx

**Comment by monsterbitar** [comment]
We are now several months into this project, there exist multiple lookup servers (written in javascript, go and c#), there exist a handful of websites around it and a small number of wallets have implemented send-support.

The main issue that is still left to be resolved is creating and agreeing on a resuable address format so that we can retain privacy while still having userfriendly names, but I still feel that implementing sending support is something that should be a higher priority.

I understand if you'd be hestitant to implementing registration support at the time being, given that address reuse is a bad practice - but ematiu from copay is working together with others to create a standardized implementation of peter todds stealth addresses that would allow us to keep the privacy.

Please do not pass up this opportunity to build the user interfaces and send-support so that bitcoin.coms wallet can be ready when the registration support gets implemented elsewhere.

**Comment by monsterbitar** [comment]
courtesy of @emergent-reasons there is tests available (https://gitlab.com/cash-accounts/specification/blob/master/TEST.md) to help ensure a good integrity when implementing cashaccount support.

If there is still anything that is missing or needing, please let us know.

**Comment by monsterbitar** [comment]
The public lookup servers are now fully feature-complete: you can register accounts with a simple POST request, lookup with a GET and the specification has been stable and peer-reviewed.

IFwallet and Badger supports sending to cashaccounts, BCHgallery and CrescentCash will be released today with both registration and sending support.

Roughly 1500 unique payment addresses have cashaccounts today, and with increased wallet support more is expected to come.

I suggest you consider building the send-support at minimum (it is a simple lookup + parse of the data, to translate from name to address).

**Comment by monsterbitar** [comment]
Since Apr 2, Electron Cash and Cashual support CashAccounts, Bitcoin Unlimited voted to build native support for CashAccounts into their full node software and work on reusable addresses (type 4 accounts) has a specification that is getting close to release which would retain privacy.

What's missing now seems to be the bitcore related wallets (bitpay, copay, bitcoincom).

Could you help provide some clarity here about your intents? Will you support cash accounts? If not, what is holding you back and what needs to be fixed for it to work for you?

**Comment by imaginaryusername** [comment]
This is especially relevant considering rest.bitcoin.com is now supporting it - any centralized or semi-centralized wallets should have little trouble just calling that.

**Comment by pokkst** [comment]
Yep. Bitcoin.com explorer lists any Cash Accounts associated with an address, IFwallet has supported sending-to for a while, BCHgallery and Crescent Cash are the first to have full support, Electron Cash is now fully supported. Tons of people use it daily, have it in their profiles/display names, etc.

Support in the Bitcoin.com wallet is needed at this point.

**Comment by GoatPharmer** [comment]
Do we have updates on bitcoin.com wallet?

**Comment by bitcoinstore** [comment]
We are busy working on releasing an all new fully native version of the Bitcoin.com wallet.  We can work on Cash Accounts after that.

**Comment by maplesyrupsucker** [comment]
We definitely have plans to support CashAccounts at some point. Like @bitcoinstore  said, the bitcoin.com wallet team is currently focusing on getting our new native mobile wallets for iOS and Android. 

The goal is to create the fastest, most user friendly wallet that will help us scale to 50M wallet users, and more importantly support innovative products, services, and community development. 

Providing support for one or many alias formats is something our engineering team regularly discusses. We're exploring all options but will prioritize after launch. Stay tuned! 🏆

**Comment by monsterbitar** [comment]
Soon to be 2 years later, you've launched the native wallet, and now you've added ethereum and unstoppable domains support. Users can now pay $40 to get an alias that functions in your app.

You can close this issue if you have no intention of doing this.
